packages/app-js/src/index.tsx

```diff
@@ -61,31 +64,56 @@ class App extends React.PureComponent<Props, State> {
       isRunning: false,
       logs: [],
       options: [],
-      snippet: this.snippets[0]
+      selected: this.snippets[0],
+      sharedExample: undefined,
+      shareLink: ''
     };
   }
 
   componentDidMount () {
+    const { location } = this.props;
+
+    const sharedExample: Snippet = location.hash.length ?
+    ({
+      code: atob(location.hash.substr(1)),
```

Comment:
We can use the router matching, i.e. https://github.com/polkadot-js/apps/blob/master/packages/app-explorer/src/BlockByHash/index.tsx#L22

Reply:
Only when using `<Switch>` and `<Route>` as far as I know https://github.com/polkadot-js/apps/blob/master/packages/app-explorer/src/index.tsx#L60

It does return an empty object in my case because I'm not rendering different Routes (as of now)...

Reply:
We can add the one with a route, most apps have that anyway. (May not have lots of different routes yet, but it has the benefit of aligning across the codebase instead of doing something special here.) No harm either way, but we can remove some basic heavy-lifting.

Reply:
true, I added it.